src/multi_record_log.rs

```diff
@@ -1,20 +1,82 @@
+use std::io;
 use std::ops::RangeBounds;
 use std::path::Path;
+use std::time::{Duration, Instant};
 
-use crate::error::{AppendError, CreateQueueError, DeleteQueueError, TruncateError};
+use crate::error::{
+    AppendError, CreateQueueError, DeleteQueueError, ReadRecordError, TruncateError,
+};
 use crate::mem;
 use crate::record::MultiPlexedRecord;
-use crate::recordlog::{ReadRecordError, RecordWriter};
+use crate::recordlog::RecordWriter;
 use crate::rolling::RollingWriter;
 
 pub struct MultiRecordLog {
     record_log_writer: crate::recordlog::RecordWriter<RollingWriter>,
     in_mem_queues: mem::MemQueues,
+    next_flush: FlushState,
+}
+
+/// Policy for flushing data
+pub enum FlushPolicy {
```

Comment:
I'm guilty of using "flush" too describe a lot different things.

Maybe `SyncPolicy` would be more accurate.
On rocksdb and leveldb that's a setting called "sync".